### PR TITLE
[[ Bug 20826 ]] Fix matrix mult constraints in doc

### PR DESCRIPTION
**Pull request by abrisan**
Fixed matrix multiplication constraints on docs

### CODE REVIEW

File: docs/dictionary/function/matrixMultiply.lcdoc [hunk 1]
```diff
@@ -45,11 +45,9 @@ second part is the column number. For example, the expression
 myArray[3,2] describes the element of myArray which is in the third row,
 second column.
 
-The number of rows in the <firstArray> must be the same as the number of
-columns in the <secondArray>, and the number of rows in the
-<secondArray> must match the number of columns in the <firstArray>.
-However, the number of rows in the <firstArray> and <secondArray> need
-not be the same.
+The number of columns in the <firstArray> must be the same as the number
```
Comment (montegoulding) [inline docs/dictionary/function/matrixMultiply.lcdoc:48]: @livecodealex I think this is actually an engine bug that the condition isn't being checked. If you look at this line https://github.com/livecode/livecode/blob/3ad55491c4ef7d7b7f2d950bb36de8e56727caf4/engine/src/exec-array.cpp#L1215 you will see `MCMatrixEntry(p_b, k, j)` expects there to be an element in the second array where row index `k` is the column index from the first array.

Comment (montegoulding) [reply docs/dictionary/function/matrixMultiply.lcdoc:48]: I think my comment above is wrong after looking at it again... sorry

Comment (abrisan) [reply docs/dictionary/function/matrixMultiply.lcdoc:48]: @montegoulding  that is correct, but that is exactly what I proposed : cols from array 1 must match rows from array 2. The rest are uncorrelated.

**Comment by montegoulding** [comment]
@livecode-vulcan review ok 52897a1

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok 52897a1c940a535d42304eb4b7814df3fa7c7827

**Comment by livecode-vulcan** [comment]
😎 test success 52897a1c940a535d42304eb4b7814df3fa7c7827

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1136)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1152)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1126)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/79)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/79)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1131)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1153)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1136)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1152)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1135)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1145)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1169)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1130)

**Pull request merged**